pkg/ipamd/ipamd.go

```diff
@@ -411,15 +430,23 @@ func (c *IPAMContext) nodeInit() error {
 
 		err = c.networkClient.UpdateRuleListBySrc(rules, srcIPNet, pbVPCcidrs, !c.networkClient.UseExternalSNAT())
 		if err != nil {
-			log.Errorf("UpdateRuleListBySrc in nodeInit() failed for IP %s: %v", info.IP, err)
+			log.Warnf("UpdateRuleListBySrc in nodeInit() failed for IP %s: %v", info.IP, err)
 		}
 	}
-	// For a new node, attach IPs
-	increasedPool, err := c.tryAssignIPs()
-	if err == nil && increasedPool {
-		c.updateLastNodeIPPoolAction()
+	return nil
+}
+
+func (c *IPAMContext) checkVPCCIDRsAndRules(oldVPCCidrs []string) []string {
```

Comment:
nit: I prefer to name functions that mutate an object's state (like this one does for the `configureIPRulesForPods()` call when the new rules don't equal the old rules) with something an action that indicates the function may change state.

Perhaps something like `updateCIDRsRulesOnChange()` might work?